app/models/task.rb

```diff
@@ -1,3 +1,14 @@
 class Task < ActiveRecord::Base
+  before_save :set_completed_field_to_false_for_new_records
+
   validates :name, presence: true
+
+  belongs_to :list
```

Comment:
I like how you added dependent destroy to user/list relation, would you consider doing the same here? Or is the task/list relationship diff?

Reply:
I have set dependent: :destroy to both User/List and List/Task models. 
list.rb has `has_many :tasks, dependent: :destroy` code in it.

Reply:
I believe both user/list and list/task should behave similarly. When there are no user, there should not be lists or tasks that belongs to the user/list.